File: Part 2 - Regression/Section 9 - Random Forest Regression/arboles aleatorios.py
# -*- coding: utf-8 -*-
"""
Created on Sat Dec 14 17:04:22 2019

@author: darub
"""
# regresion con arboles aleatorios

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


dataset = pd.read_csv("Position_Salaries.csv")

x = dataset.iloc[:,1:2].values
y = dataset.iloc[:,-1].values


# Dividir entrenamiento y testeo
# son pocos datos no se usa

# crear un objeto de arboles aleatorios
from sklearn.ensemble import RandomForestRegressor
# n_estimators es el numero de arboles, random_state semilla 0
randon_forest = RandomForestRegressor(n_estimators=300, random_state=0)
randon_forest.fit(x,y)

y_pred = randon_forest.predict([[6.5]])



# Graficar arbol
x_grid = np.arange(min(x), max(x), 0.1)
x_grid = x_grid.reshape(len(x_grid),1)

plt.scatter(x,y, color="red")
plt.plot(x_grid, randon_forest.predict(x_grid), color="blue")
plt.title("Arboles aleatorios")
plt.xlabel("Posicion del empleado")
plt.ylabel("Sueldo en ($)")
plt.show()

File: Part 2 - Regression/Section 9 - Random Forest Regression/arboles aleatorios.R
#install.packages("randomForest")
library(randomForest)
library(ggplot2)

# Regresion con arboles aleatorios

dataset = read.csv("Position_Salaries.csv")

dataset = dataset[,2:3]
# semilla 
set.seed(1234)
regression = randomForest(x = dataset[1], y = dataset$Salary,
                          ntree = 500)

# prediccion de nuevos datos
y_pred = predict(regression, newdata = data.frame(Level=6.5))


x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)

ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = "red") +
  geom_line(aes(x= x_grid, y = predict(regression, newdata = data.frame(Level=x_grid)))
            ,color="blue")+
  ggtitle("Predccion random forest")+
  xlab("Nivel del empleado")+
  ylab("Sueldo en $")











